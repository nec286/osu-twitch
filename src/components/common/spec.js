import Inferno from 'inferno';
import { renderIntoDocument, renderToSnapshot, scryRenderedDOMElementsWithTag } from 'inferno-test-utils';
import * as common from './';

describe('Badge', () => {
  it('renders OK', () => {
    const tree = renderToSnapshot(
      <common.Badge label="" value={ 999999 } />
    );
    expect(tree).toMatchSnapshot();
  })
});

describe('BeatMapLink', () => {
  it('renders OK', () => {
    const tree = renderToSnapshot(
      <common.BeatMapLink beatMap={{
        beatmap_id: 1,
        mode: 0,
        title: 'foo',
        version: 'bar'
      }} />
    );
    expect(tree).toMatchSnapshot();
  })
});

describe('FlagIcon', () => {
  it('renders OK', () => {
    const tree = renderToSnapshot(
      <common.FlagIcon country="JP" />
    );
    expect(tree).toMatchSnapshot();
  });

  it('has the correct class', () => {
    const tree = renderIntoDocument(
      <common.FlagIcon country="JP" />
    );
    const i = scryRenderedDOMElementsWithTag(tree, 'i');
    expect(i[0].className).toBe('flag-icon flag-icon-jp');
  });
});

describe('ModeSelect', () => {
  it('renders OK', () => {
    const tree = renderToSnapshot(
      <common.ModeSelect onChange={ null } />
    );
    expect(tree).toMatchSnapshot();
  });
});
