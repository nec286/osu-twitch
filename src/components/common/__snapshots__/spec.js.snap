// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Badge renders OK 1`] = `
<div
  label=""
  value={999999}
/>
`;

exports[`BeatMapLink renders OK 1`] = `
<a
  beatMap={
    Object {
      "beatmap_id": 1,
      "mode": 0,
      "title": "foo",
      "version": "bar",
    }
  }
/>
`;

exports[`FlagIcon renders OK 1`] = `
<i
  country="JP"
/>
`;

exports[`ModeSelect renders OK 1`] = `
<select
  onChange={null}
/>
`;

exports[`Statistic renders OK 1`] = `
<div
  label=""
  value={999999}
/>
`;
